src/SDDPoptimize.jl

```diff
@@ -208,42 +209,100 @@ function optimize(model::SPModel,
 
     stopping_test::Bool = false
     iteration_count::Int64 = 0
-    n::Int64 = param.forwardPassNumber
-
-
 
     while (iteration_count < param.maxItNumber) & (~stopping_test)
+        # Time execution of current pass:
+        tic()
+
         # Build given number of scenarios according to distribution
         # law specified in model.noises:
         aleas = simulate_scenarios(model.noises ,
                                     (model.stageNumber,
                                      param.forwardPassNumber,
                                      model.dimNoises))
+
+        # Forward pass
         costs, stockTrajectories, _ = forward_simulations(model,
                             param,
                             V,
                             problems,
-                            n,
                             aleas)
 
-        V0 = backward_pass(model,
+        # Backward pass
+        V0 = backward_pass!(model,
                       param,
                       V,
                       problems,
                       stockTrajectories,
                       model.noises)
 
-        iteration_count+=1;
-
+        iteration_count += 1
         upb = upper_bound(costs)
+        time = toq()
+
         if display
             println("Pass number ", iteration_count,
-                    "  Upper bound: ", upb,
-                    "  V0: ", V0)
+                    "\tEstimation of upper-bound: ", upb,
+                    "\tV0: ", V0,
+                    "\tTime: ", time)
         end
 
-        stopping_test = test_stopping_criterion(V0, upb, param.sensibility)
+    end
+
+    if display
+        println("Estimate upper-bound with Monte-Carlo ...")
+        upb = estimate_upper_bound(model, param, V, problems)
+        println("Estimation of upper-bound: ", upb,
+                "\tV0: ", V0,
+                "\t Gap (\%): ", (V0-upb)/V0)
     end
 
     return V, problems
 end
+
+
+"""
+Estimate upper bound with Monte Carlo.
+
+Parameters:
+- model (SPmodel)
+    the stochastic problem we want to optimize
+
+- param (SDDPparameters)
+    the parameters of the SDDP algorithm
+
+- V (Array{PolyhedralFunction})
+    the current estimation of Bellman's functions
+
+- problems (Array{JuMP.Model})
+    Linear model used to approximate each value function
+
+- n_simulation (Float64)
+    Number of scenarios to use to compute Monte-Carlo estimation
+
+
+Return:
+Float64 (estimation of the upper bound)
+
+"""
+function estimate_upper_bound(model, param, V, problems, n_simulation=1000)
+
+    n_fpn = param.forwardPassNumber
+    param.forwardPassNumber = n_simulation
```

Comment:
Are we sure it wont change the param outside of this function ?
 I mean if I want to do 1 forward scenario at a time and 
 call estimate_upper_bound at some point it will still be doing 1 forward scenario afterwards ?